builtin/providers/ultradns/resource_ultradns_tcpool.go

```diff
@@ -322,10 +324,26 @@ func zipRData(rds []string, rdis []udnssdk.SBRDataInfo) []map[string]interface{}
 // makeSetFromRdatas encodes an array of Rdata into a
 // *schema.Set in the appropriate structure for the schema
 func makeSetFromRdata(rds []string, rdis []udnssdk.SBRDataInfo) *schema.Set {
-	s := &schema.Set{F: hashRdatas}
+	s := &schema.Set{F: hashRdataTcpool}
 	rs := zipRData(rds, rdis)
 	for _, r := range rs {
 		s.Add(r)
 	}
 	return s
 }
+
+// hashRdataTcpool generates a hashcode for an Rdata block from tcpools
+func hashRdataTcpool(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["host"].(string)))
```

Comment:
None of these can be nil?

Reply:
All values from here should come from defaults, found in the Schema definitions above.